pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -277,13 +276,53 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 	}
 
 	cidrStrings := make([]string, 0)
-	var northInterfaces networkv1.NorthInterfacesAnnotation
-	var additionalNodeNetworks networkv1.MultiNetworkAnnotation
 
 	if len(instance.NetworkInterfaces) == 0 || (len(instance.NetworkInterfaces) == 1 && len(instance.NetworkInterfaces[0].AliasIpRanges) == 0) {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
 		return fmt.Errorf("failed to allocate cidr: Node %v has no ranges from which CIDRs can be allocated", node.Name)
 	}
+
+	defer func() {
+		if rerr != nil {
+			return
+		}
+		// update node's spec
+		if !reflect.DeepEqual(node.Spec, oldNode.Spec) {
+			for i := 0; i < cidrUpdateRetries; i++ {
+				if err = utilnode.PatchNodeCIDRs(ca.client, types.NodeName(node.Name), node.Spec.PodCIDRs); err == nil {
+					klog.InfoS("Set the node PodCIDRs", "nodeName", node.Name, "cidrStrings", node.Spec.PodCIDRs)
+					break
```

Comment:
that is why we have a controller with a workqueue, if the operation fails we return an error in the sync loop and the worker logic re-adds the failed object to the workqueue so all the sync logic is retried

Reply:
anyway, I see that this was already this way , so the comment does not really applies

Reply:
I don't understand the motivation for this change, it seems that it does the same but using a defer, that is way more complex to think about

Reply:
explained above.